File: src/main/java/br/com/blu/bs2/sintetizador/contas/csv/utils/RandomicoUtil.java
package br.com.blu.bs2.sintetizador.contas.csv.utils;

import java.util.Random;

/**
 * @author Daniel Santos
 */
public final class RandomicoUtil {

    private static final int LIMITE_MAX_RANDOMICO_INTEIRO = 500000;

    private RandomicoUtil() {
    }

    private static Random getRandom() {
        return new Random();
    }

    public static int gerarValorRandomico() {
        int min = 1;
        return min + getRandom().nextInt(LIMITE_MAX_RANDOMICO_INTEIRO);
    }

    public static Long gerarValorRandomicoLong() {
        return (long) gerarValorRandomico();
    }
}

File: src/main/kotlin/br/com/blu/bs2/sintetizador/contas/csv/utils/RandomicoUtil.kt
package br.com.blu.bs2.sintetizador.contas.csv.utils

import java.util.*

/**
 * @author Daniel Santos
 */
object RandomicoUtil {

    private const val LIMITE_MAX_RANDOMICO_INTEIRO = 500000

    private val random: Random
        get() = Random()

    fun gerarValorRandomico(): Int {
        val min = 1
        return min + random.nextInt(LIMITE_MAX_RANDOMICO_INTEIRO)
    }

    fun gerarValorRandomicoLong(): Long {
        return gerarValorRandomico().toLong()
    }
}
